website/static/js/projectSettings.js

```diff
@@ -8,77 +8,114 @@ var $osf = require('js/osfHelpers');
             self.changeMessage(self.UPDATE_SUCCESS_MESSAGE, self.MESSAGE_SUCCESS_CLASS);
-            self.category(newcategory);
-            self.dirty(false);
         },
-        updateError: function(xhr, status, error) {
+
+        /*error handlers*/
+        updateCategoryError: function(xhr, status, error) {
             var self = this;
-            self.changeMessage(self.UPDATE_ERROR_MESSAGE, self.MESSAGE_ERROR_CLASS);
-            Raven.captureMessage(self.UPDATE_ERROR_MESSAGE_RAVEN, {
+            self.changeMessage(self.UPDATE_CATEGORY_ERROR_MESSAGE, self.MESSAGE_ERROR_CLASS);
+            Raven.captureMessage(self.UPDATE_CATEGORY_ERROR_MESSAGE_RAVEN, {
                 url: self.updateUrl,
                 textStatus: status,
                 err: error
             });
         },
+        updateDescriptionError: function() {
+            var self = this;
+            self.changeMessage(self.UPDATE_DESCRIPTION_ERROR_MESSAGE, self.MESSAGE_ERROR_CLASS);
+        },
+
+        /*update handlers*/
         updateCategory: function() {
             var self = this;
             return $osf.putJSON(self.updateUrl, {
                     category: self.selectedCategory()
                 })
                 .then(function(response) {
+                    self.category = self.selectedCategory();
                     return response.updated_fields.category;
                 })
-                .done(self.updateSuccess.bind(self))
-                .fail(self.updateError.bind(self));
+                .done(self.updateTitle.bind(self))
+                .fail(self.updateCategoryError.bind(self));
+        },
+        updateTitle: function() {
+            var self = this;
+            return $osf.putJSON(self.updateUrl, {
+                    title: self.title()
+                })
+                .done(function() {
+                    self.decodedTitle = self.title();
+                },
+                self.updateDescription.bind(self));
         },
-        cancelUpdateCategory: function() {
+        updateDescription: function() {
+            var self = this;
+            return $osf.putJSON(self.updateUrl, {
+                    description: self.description()
+                })
+                .done(function() {
+                    self.decodedDescription = self.description();
+                },
+                self.updateSuccess.bind(self))
+                .fail(self.updateDescriptionError.bind(self));
+        },
+
+        /*cancel handler*/
+        cancelAll: function() {
             var self = this;
-            self.selectedCategory(self.category());
-            self.dirty(false);
+            self.selectedCategory(self.category);
+            self.title(self.decodedTitle);
+            self.description(self.decodedDescription);
             self.resetMessage();
         }
     });
 
 var ProjectSettings = {
-    NodeCategorySettings: NodeCategorySettings
+    ProjectSettings: ProjectSettings
```

Comment:
Something like changing line 156 to be a declaration (rather than adding it to an object), and line 185 to be assignment directly...

```javascript
var getConfirmationCode() ...

module.exports = {
    ProjectSettings: ProjectSettings,
    getConfirmationCode: getConfirmationCode
}